pkg/services/ngalert/state/persister_async.go

```diff
@@ -0,0 +1,62 @@
+package state
+
+import (
+	"context"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+)
+
+type AsyncStatePersister struct {
+	log log.Logger
+	// doNotSaveNormalState controls whether eval.Normal state is persisted to the database and returned by get methods.
+	doNotSaveNormalState bool
+	store                InstanceStore
+}
+
+func NewAsyncStatePersisiter(log log.Logger, cfg ManagerCfg) StatePersister {
+	return &AsyncStatePersister{
+		log:                  log,
+		store:                cfg.InstanceStore,
+		doNotSaveNormalState: cfg.DoNotSaveNormalState,
+	}
+}
+
+func (a *AsyncStatePersister) Async(ctx context.Context, ticker *clock.Ticker, cache *cache) {
+infLoop:
```

Comment:
Personal nit but I much prefer the use of `defer` than label breaks. You can't forget to stop the ticker using defer either if someone add more breaks later.

```suggestion
defer func() {
    ticker.Stop()
    a.log.Info("State sync shut down")
}()
    
```

Reply:
Agree, changed it.